app/models/active_record/sqlite/index.rb

```diff
@@ -0,0 +1,23 @@
+module ActiveRecord::SQLite::Index
+  extend ActiveSupport::Concern
+
+  included do
+    self.ignored_columns = [table_name.to_sym, :rank] # Rails parses our virtual table with these extra non-attributes.
+
+    attribute :rowid, :integer
+    alias_attribute :id, :rowid
+    self.primary_key = :rowid
+
+    # Active Record doesn't pick up the `rowid` primary key column.
+    # So we have have explicitly declare this scope to have `rowid` populated in the result set.
+    default_scope { select("#{table_name}.rowid, #{table_name}.*") }
+  end
+
+  private
+
+  def attributes_for_create(attribute_names)
```

Comment:
We don't need to override `attributes_for_update`, because the `rowid` is included in the `query_constraints_hash` method, that Active Record calls in `_update_record` or wherever it was. The details are a little fuzzy to me now because of all the details I had to do.